src/main/java/org/sonarsource/sonarlint/ls/connected/ProjectBindingManager.java

```diff
@@ -626,6 +590,13 @@ public void setBranchResolver(Function<URI, String> getReferenceBranchNameForFol
     this.getReferenceBranchNameForFolder = getReferenceBranchNameForFolder;
   }
 
+  public String resolveBranchNameForFolder(URI folder) {
+    if (getReferenceBranchNameForFolder == null) {
+      return null;
+    }
```

Comment:
I'm not sure if this check is necessary. When could this happen ?

Reply:
Also the name `getReferenceBranchNameForFolder` could be a bit confusing. It's not a getter, maybe a `supplier` ?